router-tests/modules_v1/my_custom_module/authorization_module.go

```diff
@@ -0,0 +1,37 @@
+package my_custom_module
+
+import (
+	"errors"
+	"context"
+
+	"github.com/wundergraph/cosmo/router/core"
+)
+
+const authorizationModuleID = "authorizationModule"
+
+type AuthorizationModule struct {
+}
+
+func (m *AuthorizationModule) MyModule() core.MyModuleInfo {
+	return core.MyModuleInfo{
+		ID: authorizationModuleID,
+		New: func() core.MyModule {
+			return &AuthorizationModule{}
+		},
+	}
+}
+
+func (m *AuthorizationModule) Provision(ctx context.Context) error {
+	return nil
+}
+
+func (m *AuthorizationModule) Cleanup() error {
+	return nil
+}
+
+func (m *AuthorizationModule) OnRouterRequest(reqContext core.RequestContext, params *core.RouterRequestParams) error {
```

Comment:
This brings me to another topic: *race conditions.* If we encourage pre/post hooks everywhere and global struct, how can users safely share data? Do we want to move forward with the previous approach and provide safe get/set methods, or should we go with a complete different approach?